File: Part 2 - Regression/Regression_Template/generic_regression_template.R
# Generic Regression Template

# Importing the dataset
dataset = read.csv('Salary_Positions.csv')

# Fitting the Regression Model to the dataset
dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
regressor = lm(formula = Salary ~ ., data = dataset)


# Visualizing the Polynominal Linear Regression Results
library(ggplot2)
ggplot() +
	geom_point(aes(x = dataset$Level , y = dataset$Salary),
			   colour = 'red') +
	geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)),
			  colour = 'blue') +
	ggtitle("Title_Goes_Here (Polynomial Linear Regression)") +
	xlab("xLabel_Goes_Here") +
	ylab("yLabel_Goes_Here")

# Predicting the new result with Regression Model
y_pred = predict(regressor, newdata = data.frame(Level = 6.5))

# Visualizing the Regression Model Results (High Resolution)
X_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)

ggplot() +
	geom_point(aes(x = dataset$Level , y = dataset$Salary),
			   colour = 'red') +
	geom_line(aes(x = X_grid, y = predict(regressor, newdata = data.frame(Level = X_grid))),
			  colour = 'blue') +
	ggtitle("Title_Goes_Here (Polynomial Linear Regression)") +
	xlab("XLabel_Goes_Here") +
	ylab("YLabel_Goes_Here")

File: Part 2 - Regression/Regression_Template/generic_regression_template.py
# -*- coding: utf-8 -*-
"""
Created on Sun Oct 13 12:14:56 2019

@author: Niharika_Cnr
"""

# Generic Regression Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:-1].values
y = dataset.iloc[:, -1].values

# Splitting the dataset into the Training set and Test set
"""from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)"""

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""

# Fitting the Regression Model to the dataset
# Create regressor here
regressor =  None   

# Prediciting new result with the Regression Model
y_pred = regressor.predict([[6.5]])

#---------------------------------------------------------------------------------------#
# Visualizing the Regression Results
plt.scatter(X, y, color = 'r')
plt.plot(X, regressor.predict(X), color = 'b')
plt.title("Plot_Title_goes_here")
plt.xlabel("XLabel_goes_here")
plt.ylabel("YLabel_goes_here")
plt.show()

# Visualizing the Regression Results (for higher resolution)
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, y, color = 'r')
plt.plot(X_grid, regressor.predict(X_grid), color = 'b')
plt.title("Plot_Title_goes_here (Higher Resolution)")
plt.xlabel("XLabel_goes_here")
plt.ylabel("YLabel_goes_here")
plt.show()
#---------------------------------------------------------------------------------------#
